sklearn/metrics/_classification.py

```diff
@@ -665,12 +673,19 @@ class labels [2]_.
         ``y1`` or ``y2`` are used.
 
     weights : {'linear', 'quadratic'}, default=None
-        Weighting type to calculate the score. `None` means no weighted;
-        "linear" means linear weighted; "quadratic" means quadratic weighted.
+        Weighting type to calculate the score. `None` means not weighted;
+        "linear" means linear weighting; "quadratic" means quadratic weighting.
 
     sample_weight : array-like of shape (n_samples,), default=None
         Sample weights.
 
+    zero_division : {"warn", 0.0, 1.0, np.nan}, default="warn"
+        Sets the return value when there is a zero division, e.g. when
+        `y1=y2={np.ones, np.zeros}`. If set to "warn", returns 0.0 input, but a
```

Comment:
Actually, you can have the tricky case that one could collect a new batch of fresh data that has a single class and you might end up with the following:

```python
cohen_kappa_score([0, 0, 0, 0], [0, 0, 0, 0])
```

In this particular case, the expected call should be:

```python
cohen_kappa_score([0, 0, 0, 0], [0, 0, 0, 0], labels=[0, 1])
```

To inform that we should have 0/1 labels. This call currently would raise `nan`. However, if we are in the setting that I describe above then we should obtain a score of 1.0 because we are perfectly classifying. So if we are able to pass `zero_division`:

```python
cohen_kappa_score([0, 0, 0, 0], [0, 0, 0, 0], labels=[0, 1], zero_division=1.0)
```

This allow to get the score that we would expect.

As you mentioned, this is really a weird use case and we should not assume that this is normal as a default and having a warning as default and providing 0.0 would be the most safe way.

Reply:
Ok, so I understand that this is to be used on an automated task, where users know from experience that batches, that previously had a nan score are indeed mostly valid and they therefor chose to declare as a `1.0` score (or `0.0` or keep `nan`).

Reply:
> where users know from experience that batches, that would have a nan score are indeed mostly valid and they therefor chose to declare as a 1.0 score

Yes exactly.